pkg/endpoint/bpf.go

```diff
@@ -602,20 +642,35 @@ func (e *Endpoint) regenerateBPF(owner Owner, epdir, reason string) (uint64, err
 			e.L3Maps.DestroyBpfMap(IPv4Egress, e.IPv4EgressMapPathLocked())
 		}
 	}
+
+	// Since the endpoint's lock will be unlocked, we need to
+	// store the current endpoint state so we can later on
+	// update the CT without requiring to lock the endpoint again.
+	isPolicyEnforced := e.IngressOrEgressIsEnforced()
+	isLocal := e.Opts.IsEnabled(OptionConntrackLocal)
+	epIPs := []net.IP{e.IPv4.IP(), e.IPv6.IP()}
+
 	e.Mutex.Unlock()
 
 	libdir := owner.GetBpfDir()
 	rundir := owner.GetStateDir()
 	debug := strconv.FormatBool(owner.DebugEnabled())
 
 	err = e.runInit(libdir, rundir, epdir, epInfoCache.ifName, debug)
-	if err == nil {
-		// The last operation hooks the endpoint into the endpoint table and exposes it
-		err = lxcmap.WriteEndpoint(epInfoCache)
-		if err != nil {
-			log.WithField(logfields.EndpointID, e.ID).WithError(err).Error("Exposing new bpf failed!")
-		}
+	if err != nil {
+		return epInfoCache.revision, err
+	}
+	// REVIEW: should the CT entry clean up also happen
```

Comment:
Seems legit to me. Also depends on whether the policy enforcement is `default` or `always` or `never`. But I think `polciyChanged` already takes that into account?